server/src/main/java/com/lunchtime/repository/OrderRepository.java

```diff
@@ -0,0 +1,41 @@
+package com.lunchtime.repository;
+
+import com.lunchtime.models.Order;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
+import org.springframework.stereotype.Repository;
+
+import java.util.Date;
+import java.util.List;
+import java.util.Optional;
+
+@Repository
+public interface OrderRepository extends JpaRepository<Order, Long> {
+
+    @Query("select o from Order o where (o.isDeleted = false) and "
+        + "(o.table.restaurant.id = :id) and "
+        + "((:start >= o.startTime and :start < o.finishTime) or "
+        + "(:finish > o.startTime and :finish < o.finishTime) or "
+        + "(:start <= o.startTime and :finish >= o.finishTime))")
+    List<Order> findAllOrdersByRestaurantIdInTime(
+        @Param("id") Long id, @Param("start") Date start, @Param("finish") Date finish);
+
```

Comment:
Can't understant line's 22 purpose. 

And why we can't use this in line 21?  `...:finish <= o.finishTime`

Reply:
Line 22 new Order - from 13:00 to 18:00. If there is exist active order from 14:00 to 17:00. Than new Order can't be created

Line 21 new Order - from 13:00 to 18:00. Exist order - from 17:00 to 18:05

Reply:
> Line 21 new Order - from 13:00 to 18:00. Exist order - from 17:00 to 18:05
may be you mean this in line 20?

> Line 22 new Order - from 13:00 to 18:00. If there is exist active order from 14:00 to 17:00. Than new Order can't be created

> Line 22 new Order - from 13:00 to 18:00. If there is exist active order from 14:00 to 17:00. Than new Order can't be created
So we need `!not` operator? Or what? :)

Reply:
> Line 21 new Order - from 13:00 to 18:00. Exist order - from 17:00 to 18:05
> may be you mean this in line 20?

start = 13:00 < (o.startTime == 17:00)
Line 21

> So we need !not operator? Or what? :)
It works well now, but you can change query and try